### ISSUE TITLE
ejabberd 21.01 before Debian 11 freeze?

### ISSUE THREAD

**Issue by Neustradamus** [processone/ejabberd#3465]
Since ejabberd 20.12 (and deps), there were some improvements about SCRAM/LDAP/Dependencies.
- https://github.com/processone/ejabberd/compare/20.12...master

I think it would be very wise to do the new release before Debian 11 freeze deadline (2021-01-12), and the next Debian will be in 2023.

The copyright update must be done before of course.

Note: I have requested an update of ejabberd_auth_http to have SCRAM sync.

Thanks in advance.

**Comment by debalance** [comment]
The relevant freeze date here is 2021-Feb-12, see https://release.debian.org

**Comment by Neustradamus** [comment]
@debalance: Thanks for your comment.

2 days ago, in another project, the Debian maintainer of the project said:
> A new version requires a transition, and the transition freeze starts 2021-01-12. We need to get clearance from the release team for a transition beforehand, and the new version will need to have migrated to testing/bullseye before the freeze starts, so you'd need to give it 2-5 days assuming no blocking bugs appear. This is probably only going to fly if a new release is published in the next couple of days, given my own time constraints.
> 
> On the other hand I don't mind refreshing the Debian patches based on what's gone upstream and otherwise cherry-picking important fixes without pushing a new upstream release. The only stipulation is that we can't have a significant ABI change (which needs a transition). That would buy us another month before the soft freeze on 2021-02-12. After that I can still get bug fixes in targeting specific Debian bugs.

If I understand perfectly for ejabberd project, after 2021-01-12, it will be a 20.12-x, instead of 21.01, no?

**Comment by debalance** [comment]
"transition" means library updates which require many other packages to be rebuilt. It does not affect ejabberd. I have always happily uploaded new versions of my packages until the soft freeze, see https://release.debian.org/bullseye/freeze_policy.html#soft

**Comment by debalance** [comment]
To be exact: a new ejabberd release must have migrated to testing before the soft freeze on 2021-Feb-12.
It takes at least 5 days after I have packaged+uploaded a new release for it to migrate, so a new ejabberd release until end of January should be fine.

**Comment by Neustradamus** [comment]
@debalance: Thanks for your clarification :)

**Comment by badlop** [comment]
ejabberd 21.01 has just been published

**Comment by Neustradamus** [comment]
@badlop: Thanks!

**Comment by debalance** [comment]
Splitting the translations into a separate repository makes things rather complicated for me...

**Comment by badlop** [comment]
The PO files are not needed for compiling, installing or running ejabberd.

They are used only for improving translated strings, or for running "make
translations"


El ds., 30 de gen. 2021, 14:27, Philipp Huebner <notifications@github.com>
va escriure:

**Comment by licaon-kter** [comment]
@debalance ^^^

I actually delete msg/* so that translations are not loaded at start, to speed up things.

**Comment by debalance** [comment]
@licaon-kter That's not an option for the official Debian package

@badlop Just to make sure I understand you correctly: the translations are still part of ejabberd in priv/msgs/*.msg, so I can go without ejabberd-po and still have the same i18n features as before?

**Comment by licaon-kter** [comment]
@debalance I just mentioned that, eg. You could pack those separately. :)

**Comment by badlop** [comment]
Right. As packager and at runtime, PO files are not needed. In fact
ejabberd doesn't understand the gettext syntax.

The only files that ejabberd uses at runtime are the *.msg files. If you
inspect them, they have erlang syntax. Those translations are loaded at
start time.

The *.po and *. pot files exist to make life easier for translators,
because those have gettext syntax and are easy to edit with several
editors.

And "make translations" runs the script that updates all of them, and
converts from ones to the others, ready to be committed to git.

El ds., 30 de gen. 2021, 15:57, Philipp Huebner <notifications@github.com>
va escriure:

**Comment by Neustradamus** [comment]
@debalance: ejabberd uses "priv/msgs/*.msg", I know that the name is not really good, I have requested to change to "translations", it has been refused: https://github.com/processone/ejabberd/issues/3498.

Note: It is not possible to add new package into Debian 11, the date was 2021-01-12.

**Comment by debalance** [comment]
This issue can be closed, ejabberd 21.01 is uploaded to unstable.

**Comment by Neustradamus** [comment]
@debalance: Thanks!
